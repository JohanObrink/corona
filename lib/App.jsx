import React, { useReducer } from 'react'
import { reducer } from './form/formReducer'
import Form from './form/Form'
import { form } from './form/data'
import { ProgressBar } from '@sebgroup/react-components/dist/ProgressBar/ProgressBar'
import api from './api'

export default () => {
  const [state, dispatch] = useReducer(reducer, { form, data: {}, currentStepIndex: 0 })
  return (
    <>
      <ProgressBar
        value={(state.currentStepIndex / (form.steps.length - 1)) * 100}
      />
      <div className='container'>
        <h1>Titel p√• verktyget</h1>
        <Form form={form} stepIndex={state.currentStepIndex} dispatch={dispatch} />
      </div>
    </>
  )
}
