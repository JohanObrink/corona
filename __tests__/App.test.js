import React from 'react'
import renderer from 'react-test-renderer'
import App from '../lib/App'
import { isExportDeclaration } from 'typescript'

describe('App', () => {
  it('renders', () => {
    const component = renderer.create(<App />)
    expect(component).toBeDefined()
  })
})
