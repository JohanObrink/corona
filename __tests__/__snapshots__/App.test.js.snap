// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the title 1`] = `
Array [
  <div
    className="custom-progress"
  >
    <div
      className="custom-progress-bar"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>,
  <div
    className="container"
  >
    <h1>
      Titel p√• verktyget
    </h1>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h3 />
      </div>
    </div>
    <hr />
    <div
      className="row justify-content-end"
    >
      <div
        className="col-auto prev-next-buttons"
      />
    </div>
  </div>,
]
`;
